### PR TITLE
canary e2e tests + Github actions 

### PR DESCRIPTION
**Pull request by henryZrncik**

### CODE REVIEW

File: test/e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+const (
+	httpUrlPrefix   = "http://localhost:8080"
+	metricsEndpoint = "/metrics"
+)
+
+/* test checks for following:
+*  the presence of canary topic,
+*  Liveness of topic (messages being produced),
+ */
+func TestCanaryTopicLivenesss(t *testing.T) {
```
Comment (ppatierno) [inline test/e2e_test.go:25]: too many "s" :-D

File: test/test_utilities.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		h.partitionsConsumptionSlice[msg.Partition] = true
+		// Check that Consumer group consumed at least one message for each partition on canary topic
+		log.Println("msg. arrived")
+		if h.isEveryPartitionConsumed() {
+			log.Println("all partitions consumed")
+			h.consumingDone <- true
+		}
+		h.mutexWritePartitionPresence.Unlock()
+		sess.MarkMessage(msg, "")
+
```
Comment (ppatierno) [inline test/test_utilities.go:52]: useless blank line

**Review by ppatierno — COMMENTED** [review]
We are almost there :-)

File: test/service_manager/service_manager.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+func (c *ServiceManager) StartCanary() {
+	log.Println("Starting Canary")
+	c.setUpCanaryParamsViaEnv()
+	myCmd := exec.Command("go", "run",  c.pathToCanaryMain )
+
+	if err := myCmd.Start(); err != nil {
+		log.Fatal(err.Error())
+	}
+	// before we start to consider canary prepared we wait for it to create expected topic
+	c.waitForCanarySetUp()
```
[the remark this answers is not in the document]
Comment (henryZrncik) [reply test/service_manager/service_manager.go]: you are right, we don't need it anymore.

**Review by ppatierno — APPROVED** [review]
Finally, lgtm! Thanks!

File: .github/workflows/acceptancetest.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+name: acceptancetest
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```
Comment (Frawless) [inline .github/workflows/acceptancetest.yml:7]:
```suggestion
    branches: 
       - "*"
```

File: .github/workflows/acceptancetest.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+name: acceptancetest
+
+on:
+  push:
+    branches: [main]
```
Comment (Frawless) [inline .github/workflows/acceptancetest.yml:5]:
```suggestion
    branches: 
       - "main"
```

File: .github/workflows/acceptancetest.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+name: acceptancetest
```
Comment (Frawless) [inline .github/workflows/acceptancetest.yml:1]: Maybe just `acceptance` is enough? @strimzi/maintainers wdyt?

Comment (ppatierno) [reply .github/workflows/acceptancetest.yml:1]: I am fine, but we should change the YAML file name as well then.

**Pull request merged**